backend/src/controllers/teamController.ts

```diff
@@ -52,46 +79,87 @@ const getTeam = async (req: Request, res: Response) => {
       },
     });
 
-    // Handle  404
+    // Handle 404
     if (!team) {
       return errorResponse(404, "Team not found", res);
     }
-    if (team.ownerId !== req.headers.owneraddress) {
+    const teamMembers = team.members.map((member) => member.address);
+    if (teamMembers.includes(String(callerAddress)) === false) {
       return errorResponse(401, "Unauthorized", res);
     }
     res.status(200).json({ team, success: true });
-  } catch (error) {
-    return errorResponse(500, error, res);
+  } catch (error: any) {
+    return errorResponse(500, error.message, res);
   }
 };
 
 // Get teams owned by user
 const getAllTeams = async (req: Request, res: Response) => {
-  /* 
+  /*
   #swagger.tags = ['Teams']
   */
-  const ownerId = String(req.headers.owneraddress);
+  const callerAddress = String(req.headers.calleraddress);
   try {
-    const teams = await prisma.team.findMany({
+    // Get teams owned by the user
+    const ownedTeams = await prisma.team.findMany({
       where: {
-        ownerId: ownerId,
+        ownerAddress: callerAddress,
+      },
+      include: {
+        members: true,
       },
     });
-    res.status(200).json({ teams, success: true });
-  } catch (error) {
-    return errorResponse(500, error, res);
+
+    // Get teams where the user is a member
+    const memberTeams = await prisma.team.findMany({
+      where: {
+        members: {
+          some: {
+            address: callerAddress,
+          },
+        },
+      },
+      include: {
+        members: true,
+      },
+    });
+
+    // Combine owned and member teams
+    const allTeams = [...ownedTeams, ...memberTeams];
+
+    // Filter out duplicate teams
+    const uniqueTeams = allTeams.filter(
+      (team, index, self) => index === self.findIndex((t) => t.id === team.id)
+    );
+
+    res.status(200).json({ teams: uniqueTeams, success: true });
+  } catch (error: any) {
+    console.log("Error:", error);
+    return errorResponse(500, error.message, res);
   }
 };
 
-// update team
+// Update team
 const updateTeam = async (req: Request, res: Response) => {
-  /* 
+  /*
   #swagger.tags = ['Teams']
   */
   const { id } = req.params;
   const { name, description, bankAddress } = req.body;
+  const callerAddress = req.headers.calleraddress;
```

Comment:
Why you get the caller address in the header ?

Reply:
Because I wanted to set a standard for all requests. Normally in GET and DELETE operations, we wouldn't be able to send a body. That's why I followed this convention of getting callerAddress in the headers.

Reply:
I'm not sure about how to do this